membership-attribute-service/app/services/AttributesMaker.scala

```diff
@@ -35,18 +44,36 @@ class AttributesMaker extends LazyLogging {
 
     val hasAttributableProduct = membershipSub.nonEmpty || contributionSub.nonEmpty || subsWhichIncludeDigitalPack.nonEmpty
 
-    hasAttributableProduct.option {
-      val tier: Option[String] = membershipSub.flatMap(getCurrentPlans(_).headOption.map(_.charges.benefits.head.id))
-      val recurringContributionPaymentPlan: Option[String] = contributionSub.flatMap(getTopPlanName)
-      val membershipJoinDate: Option[LocalDate] = membershipSub.map(_.startDate)
-      val latestDigitalPackExpiryDate: Option[LocalDate] = Some(subsWhichIncludeDigitalPack.map(_.termEndDate)).filter(_.nonEmpty).map(_.max)
-      ZuoraAttributes(
-        UserId = identityId,
-        Tier = tier,
-        RecurringContributionPaymentPlan = recurringContributionPaymentPlan,
-        MembershipJoinDate = membershipJoinDate,
-        DigitalSubscriptionExpiryDate = latestDigitalPackExpiryDate
-      )
+    val maybeMembershipSubAndAccount = membershipSub flatMap { sub: Subscription[AnyPlan] => subsWithAccounts.find(subWithAccount => subWithAccount.subscription == Some(sub)) }
+
+    val maybeActionAvailable = maybeMembershipSubAndAccount flatMap { subAndAccount: CustomerAccount => subAndAccount.subscription map { sub: Subscription[AnyPlan] =>
+        actionAvailableFor(subAndAccount.account, sub, paymentMethodGetter) map { actionAvailable: Boolean =>
+          if(actionAvailable) Some("membership")
```

Comment:
Could we use something more specific like Some("membership_card_update")?

Reply:
Ignore this... I just saw that it's called 'ActionAvailableFor' and this makes more sense now.